pkg/icingadb/ha.go

```diff
@@ -274,15 +287,25 @@ func (h *HA) realize(ctx context.Context, s *icingaredisv1.IcingaStatus, t *type
 			switch errQuery {
 			case nil:
 				otherResponsible = true
-				if shouldLog {
+				if shouldLogRoutineEvents {
 					h.logger.Infow("Another instance is active",
 						zap.String("instance_id", instance.Id.String()),
 						zap.String("environment", envId.String()),
-						"heartbeat", instance.Heartbeat,
+						zap.Time("heartbeat", instance.Heartbeat.Time()),
 						zap.Duration("heartbeat_age", time.Since(instance.Heartbeat.Time())))
 				}
 			case sql.ErrNoRows:
-				takeover = true
+				fields := []any{
+					zap.String("instance_id", instance.Id.String()),
```

Comment:
When the SQL query error type matches `sql.errNoRows`, it won't scan any rows into the `instance` variable and thus `instance.Id` will always be empty.